src/ftest.jl

```diff
@@ -0,0 +1,140 @@
+to the data
+
+# Examples:
+Suppose we want to compare the effects of two or more treatments on some result. Because
+this is an ANOVA, our null hypothesis is that Result~1 fits the data as well as
+Result~Treatment. 
+```jldoctest
+julia> dat = DataFrame(Treatment=[1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2.],
+                     Result=[1.1, 1.2, 1, 2.2, 1.9, 2, .9, 1, 1, 2.2, 2, 2]);
+julia> mod = lm(@formula(Result~Treatment), dat        );
+julia> nullmod = lm(@formula(Result~1), d);
+julia> ft = ftest(mod.model, nullmod.model)
+        Res. DOF DOF ΔDOF    SSR    ΔSSR      R²    ΔR²       F* p(>F)
+Model 1       10   3      0.1283          0.9603                      
+Model 2       11   2   -1 3.2292 -3.1008 -0.0000 0.9603 241.6234 <1e-7
+```
+"""
+function ftest(mods::LinearModel...)
+    nmodels = length(mods)
+    for i in 2:nmodels
+        issubmodel(mods[i], mods[i-1]) || 
+        throw(ArgumentError("F test $i is only valid if model $i is nested in model $i-1"))
+    end
+
+    SSR = deviance.(mods)
+
+    nparams = dof.(mods)
+
+    df2 = _diffn(nparams)
+    df1 = Int.(dof_residual.(mods))
+
+    MSR1 = dividetuples(_diff(SSR), df2)
+    MSR2 = dividetuples(SSR, df1)[1:nmodels-1]
+
+    fstat = dividetuples(MSR1, MSR2)
+    pval = PValue.(ccdf.(FDist.(df2, df1[1:nmodels-1]), fstat))
+    return FTestResult(SSR, nparams, df1, r2.(mods), fstat, pval)
+end
+
+function show{N}(io::IO, ftr::FTestResult{N})
+    Δdof = _diffn(ftr.dof_resid)
+    Δssr = _diffn(ftr.ssr)
+    ΔR² = _diffn(ftr.r2)
+
+    nc = 10
+    nr = N
+    outrows = Matrix{String}(nr+1, nc)
+    
+    outrows[1, :] = ["", "Res. DOF", "DOF", "ΔDOF", "SSR", "ΔSSR",
+                     "R²", "ΔR²", "F*", "p(>F)"]
+
+    outrows[2, :] = ["Model 1", @sprintf("%.0d", ftr.dof_resid[1]),
+                     @sprintf("%.0d", ftr.dof[1]), " ",
+                     @sprintf("%.4f", ftr.ssr[1]), " ",
+                     @sprintf("%.4f", ftr.r2[1]), " ", " ", " "]
+    
+    for i in 2:nr
+        outrows[i+1, :] = ["Model $i", @sprintf("%.0d", ftr.dof_resid[i]),
+                           @sprintf("%.0d", ftr.dof[i]), @sprintf("%.0d", Δdof[i-1]),
+                           @sprintf("%.4f", ftr.ssr[i]), @sprintf("%.4f", Δssr[i-1]),
+                           @sprintf("%.4f", ftr.r2[i]), @sprintf("%.4f", ΔR²[i-1]),
+                           @sprintf("%.4f", ftr.fstat[i-1]), string(ftr.pval[i-1]) ]
+    end
+    colwidths = length.(outrows)
+    max_colwidths = [maximum(view(colwidths, :, i)) for i in 1:nc]
+
+    for r in 1:nr+1
+        for c in 1:nc
+            cur_cell = outrows[r, c]
+            cur_cell_len = length(cur_cell)
+            
+            padding = " "^(max_colwidths[c]-cur_cell_len)
+            if c > 1 
+                padding = " "*padding
+            end
+            
+            print(io, padding)
+            print(io, cur_cell)
+        end
+        print(io, "\n")
+    end
+end
+
```

Comment:
Since it looks like we'll need another round of fixes, it would be nice to remove these two empty lines in your next (and hopefully last) commit.